ftbfs regression between beta2 and rc1

Hydrogen version *          : 1.0.0-rc1
Operating system + version  :  Debian sid (current as of 2020-06-16)
Audio driver + version      : NA

Thank you for the work towards a new release!  I'm working to reintroduce Hydrogen to Debian (and derivatives), because it was dropped when Qt4 was dropped.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```
…
[ 89%] Built target hydrogen
In file included from /usr/include/c++/9/cassert:44,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/logger.h:26,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/object.h:26,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/midi_action.h:24,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/hydrogen.h:27,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:9,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/logger.h: In static member function 'static H2Core::Logger* H2Core::Logger::get_instance()':
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/logger.h:75:34: error: '__assert_fail' was not declared in this scope; did you mean 'lo::__assert_fail'?
   75 |   static Logger* get_instance(){ assert(__instance); return __instance; }
      |                                  ^~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /usr/include/lo/lo_cpp.h:128,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:8,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/usr/include/assert.h:69:13: note: 'lo::__assert_fail' declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/midi_action.h:27,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/hydrogen.h:27,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:9,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/midi_action.h: In static member function 'static MidiActionManager* MidiActionManager::get_instance()':
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/midi_action.h:284:46: error: '__assert_fail' was not declared in this scope; did you mean 'lo::__assert_fail'?
  284 |   static MidiActionManager* get_instance() { assert(__instance); return __instance; }
      |                                              ^~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /usr/include/lo/lo_cpp.h:128,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:8,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/usr/include/assert.h:69:13: note: 'lo::__assert_fail' declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/hydrogen.h:37,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:9,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/hydrogen.h: In static member function 'static H2Core::Hydrogen* H2Core::Hydrogen::get_instance()':
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/hydrogen.h:105:35: error: '__assert_fail' was not declared in this scope; did you mean 'lo::__assert_fail'?
  105 |  static Hydrogen* get_instance(){ assert(__instance); return __instance; };
      |                                   ^~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /usr/include/lo/lo_cpp.h:128,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:8,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/usr/include/assert.h:69:13: note: 'lo::__assert_fail' declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/Preferences.h:28,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:6:
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/Preferences.h: In static member function 'static H2Core::Preferences* H2Core::Preferences::get_instance()':
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/Preferences.h:367:39: error: '__assert_fail' was not declared in this scope; did you mean 'lo::__assert_fail'?
  367 |  static Preferences*  get_instance(){ assert(__instance); return __instance; }
      |                                       ^~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /usr/include/lo/lo_cpp.h:128,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:8,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/usr/include/assert.h:69:13: note: 'lo::__assert_fail' declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/osc_server.h:33,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:7:
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/osc_server.h: In static member function 'static OscServer* OscServer::get_instance()':
/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/core/include/hydrogen/osc_server.h:116:38: error: '__assert_fail' was not declared in this scope; did you mean 'lo::__assert_fail'?
  116 |   static OscServer* get_instance() { assert(__instance); return __instance; }
      |                                      ^~~~~~
In file included from /usr/include/c++/9/cassert:44,
                 from /usr/include/lo/lo_cpp.h:128,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.h:8,
                 from /build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/src/tests/osc_server_test.cpp:5:
/usr/include/assert.h:69:13: note: 'lo::__assert_fail' declared here
   69 | extern void __assert_fail (const char *__assertion, const char *__file,
      |             ^~~~~~~~~~~~~
make[2]: *** [src/tests/CMakeFiles/tests.dir/build.make:222: src/tests/CMakeFiles/tests.dir/osc_server_test.cpp.o] Error 1
make[2]: Leaving directory '/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/obj-x86_64-linux-gnu'
make[1]: *** [CMakeFiles/Makefile2:318: src/tests/CMakeFiles/tests.dir/all] Error 2
make[1]: Leaving directory '/build/hydrogen-1EeZyB/hydrogen-1.0.0~rc1/obj-x86_64-linux-gnu'
make: *** [Makefile:155: all] Error 2
```

This problem does not exist in beta2.  Full build logs and build info available on request.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

A short list of relevant deps is: liblo-dev=0.31, cmake=3.16.3, gcc=9.2.1.

--- a/src/tests/osc_server_test.h
+++ b/src/tests/osc_server_test.h
@@ -4,9 +4,9 @@
 
 #include <cppunit/extensions/HelperMacros.h>
 
+#include <hydrogen/hydrogen.h>
 #include <lo/lo.h>
 #include <lo/lo_cpp.h>
-#include <hydrogen/hydrogen.h>
 
 class OscServerTest : public CppUnit::TestFixture {
 	CPPUNIT_TEST_SUITE( OscServerTest );

